app/environments/brassbirmingham/brassbirmingham/envs/classes/player.py

```diff
@@ -114,7 +120,19 @@ def canPlaceBuilding(
         if building.onlyPhaseTwo and self.board.era != Era.railroad:
             return False
 
-        return buildLocation.isPossibleBuild(building)
+        isValidBuild = False        
+        for b in self.buildings:
+            if b.name == building.name:
+                if building.town is None and b.tier < building.tier:
```

Comment:
```suggestion
                if not building.isActive and b.tier < building.tier:
```

Reply:
`isActive` gets set to false when a market building is sold. So that would break this, or that's a bug

Reply:
Or when overbuilding for that matter